source/tests/system/nidmm_driver_api_tests.cpp

```diff
@@ -269,6 +336,67 @@ TEST_F(NiDmmDriverApiTest, ConfigureCurrentSourse_CompletesSuccessfully)
     expect_api_success(response.status());
 }
 
+TEST_F(NiDmmDriverApiTest, ConfiguredCurrentSource_ExportAndReset_ImportsSuccessfully)
+{
+    ConfigureCurrentSource(0.0001);
+    
+    auto exportedConfigurationResponse = ExportConfiguration();
+    Reset();
+
+    ImportCoinfiguration(exportedConfigurationResponse);
+    ValidateCurrentSource(0.0001);
+}
+
+TEST_F(NiDmmDriverApiTest, ConfiguredTrigger_ConfiguresSuccessfully)
+{
+    ::grpc::ClientContext configureTriggerContext;
+    dmm::ConfigureTriggerRequest configureTriggerRequest;
+    configureTriggerRequest.mutable_vi()->set_id(GetSessionId());
+    configureTriggerRequest.set_trigger_source(dmm::TriggerSource::TRIGGER_SOURCE_NIDMM_VAL_SOFTWARE_TRIG);
+    configureTriggerRequest.set_trigger_delay(-1);
+    dmm::ConfigureTriggerResponse configureTriggerResponse;
+    
+    ::grpc::Status configureTriggerStatus = GetStub()->ConfigureTrigger(&configureTriggerContext, configureTriggerRequest, &configureTriggerResponse);
+
+    EXPECT_TRUE(configureTriggerStatus.ok());
+    expect_api_success(configureTriggerResponse.status());
+}
+
+TEST_F(NiDmmDriverApiTest, AcquireMeasurement_CompletesSuccesfully)
+{
+    ::grpc::ClientContext context;
+    dmm::ReadRequest request;
+    request.mutable_vi()->set_id(GetSessionId());
+    request.set_maximum_time(1000);
+    dmm::ReadResponse response;
+    ::grpc::Status status = GetStub()->Read(&context, request, &response);
+
+    EXPECT_TRUE(status.ok());
+    expect_api_success(response.status());
+    EXPECT_NE(0, response.reading());
+}
+
+TEST_F(NiDmmDriverApiTest, SelfCalibrate_CompletesSuccessfully)
+{
+  ::grpc::ClientContext selfCalSupportedContext;
+  dmm::GetSelfCalSupportedRequest selfCalSupportedRequest;
+  selfCalSupportedRequest.mutable_vi()->set_id(GetSessionId());
+  dmm::GetSelfCalSupportedResponse selfCalSupportedResponse;
+  auto selfCalSupportedStatus = GetStub()->GetSelfCalSupported(&selfCalSupportedContext, selfCalSupportedRequest, &selfCalSupportedResponse);
+
+  bool supported = selfCalSupportedResponse.self_cal_supported();
+  if (supported) {
```

Comment:
This is problematic. If we run the test against simulated device that doesn't support self-calibration, this test won't test if self-calibration works properly and still pass. What if we fail the test with error message saying "self calibration not support, run test against device that supports self calibration?

Reply:
i removed the if statement